workflows/scnanoseq.nf

```diff
@@ -1,15 +1,109 @@
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    IMPORT MODULES / SUBWORKFLOWS / FUNCTIONS
+    PARAMETER PRESETS
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-include { FASTQC                 } from '../modules/nf-core/fastqc/main'
-include { MULTIQC                } from '../modules/nf-core/multiqc/main'
-include { paramsSummaryMap       } from 'plugin/nf-validation'
-include { paramsSummaryMultiqc   } from '../subworkflows/nf-core/utils_nfcore_pipeline'
-include { softwareVersionsToYAML } from '../subworkflows/nf-core/utils_nfcore_pipeline'
-include { methodsDescriptionText } from '../subworkflows/local/utils_nfcore_scnanoseq_pipeline'
+
+if (params.barcode_format.equals("10X_3v3")) {
+    blaze_whitelist = file("$baseDir/assets/whitelist/3M-february-2018.zip")
+}
+else{
+    blaze_whitelist = file("$baseDir/assets/whitelist/737K-august-2016.txt.zip")
+}
+
+if (params.whitelist) {
+    blaze_whitelist = params.whitelist
```

Comment:
maybe you need to check that either `barcode_format` or `whitelist` is set? otherwise this could overwrite some user settings?

Reply:
We do actually want to override the whitelist regardless if `barcode_format` is set. A user could potentially have a different set of barcodes to search for despite using one of the `barcode_format` options, as `barcode_format` is additionally used by BLAZE to find barcodes in a read.

Although to your point, it may make sense to combine the two if statements like below?

```
if (params.whitelist) {
    blaze_whitelist = params.whitelist
}
else {
    if (params.barcode_format.equals("10X_3v3")) {
        blaze_whitelist = file("$baseDir/assets/whitelist/3M-february-2018.zip")
    }   
    else if (params.barcode_format.equals("10X_5v2") {
        blaze_whitelist = file("$baseDir/assets/whitelist/737K-august-2016.txt.zip")
    }   
}

```

Reply:
yes, looks good. also mention maybe in the help text that one parameter overwrites the other

Reply:
Yep all done with commit c082e20